### PR TITLE
Feature/error handling

### PR DESCRIPTION
**Pull request by tylrs**
## Description

- Displays error messages if api calls fail (POST and all GETs) or if no rooms are available on a particular date
- Still need to add error handling for login info
- Obviously a lot of styling left to be done as well 

## Questions for Reviewer

- Any places to refactor?
- Should error messages be changed or made more specific? 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Refactor

## Checklist:
- [X] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -48,14 +49,22 @@ function validateLogin() {
   event.preventDefault();
```
Comment (alexmkio) [inline src/scripts.js:49]: pass in event through the function

File: src/scripts.js [hunk 2]
```diff
@@ -48,14 +49,22 @@ function validateLogin() {
...
   if (userName.includes('customer')) {
     fetchCustomer(userName.split('r')[1])
     .then(data => {
       let customer = new Customer(data)
       validatePassword(customer, password);
     })
     .catch(error => {
-      console.log(error);
+      console.log('error')
+      let message = `Error please try logging in again!`
```
Comment (alexmkio) [inline src/scripts.js:62]: I got this message when using correct credentials when I turned the api off. Could be more specific and let the user know it wasn't a password validation issue.

File: src/apiCalls.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ export const fetchApiData = (type) => {
   return fetch(`http://localhost:3001/api/v1/${type}`)
     .then(response => {
       if (!response.ok) {
```
Comment (alexmkio) [inline src/apiCalls.js:4]: I think both of us could DRY up these conditionals by creating a checkForError function like [here](https://github.com/zachjjohns/travel-tracker/blob/main/src/API.js)

**Review by alexmkio — COMMENTED** [review]
When I try to login (customer50, overlook2021) I'm getting the following console error:
Assertion failed: Input argument is not an HTMLInputElement

I then tried to console hotel and am getting undefined. Thoughts?

File: src/domUpdates.js [hunk 4]
```diff
@@ -122,6 +122,13 @@ let domUpdates = {
     `
   },
 
+  displayMessage(element, message) {
```
Comment (alexmkio) [inline src/domUpdates.js:125]: this is a sexy function

File: src/scripts.js (hunk 2 above)
Comment (tylrs) [reply src/scripts.js:62]: Yes I agree, I'll make that more specific!

File: src/scripts.js (hunk 1 above)
Comment (tylrs) [reply src/scripts.js:49]: Yep good idea!

**Comment by tylrs** [comment]
I realized that my password is overlook2020 and not overlook2021 lol

**Comment by alexmkio** [comment]
I was able to login as a customer using the new password and:

1. see previously booked rooms
2. the total I've spent
3. book a room and receive confirmation of the booking
4. and I received a message when there are no rooms available for a given day

Nice work!

**Pull request merged**